Test/Get-WhiskeyPowerShellModule.Tests.ps1

```diff
@@ -0,0 +1,128 @@
+    $script:context = New-WhiskeyTestContext -ForDeveloper
+}
+
+function GivenModule
+{
+    $script:taskParameter['Name'] = $moduleName
+}
+
+function GivenVersion 
+{
+    $script:taskParameter['Version'] = $version
+}
+
+function GivenInvalidVersion 
+{
+    $script:taskParameter['Version'] = $badVersion
+}
+
+function GivenInvalidModule 
+{
+    $script:taskParameter['Name'] = $invalidModuleName
+}
+
+function GivenCleanMode
+{
+    $script:context.Runmode = 'clean'
+}
+function WhenPowershellModuleIsRan
+{
+    $global:Error.clear()
+    try
+    {
+        Get-whiskeyPowershellModule -TaskContext $context -TaskParameter $taskParameter
+    }
+    catch
+    {
+        write-error $_
+    }
+}
+
+function ThenModuleShouldBeInstalled
+{
+    $modulePath = Join-path -Path 'Modules' -ChildPath $moduleName
+    it 'should have installed the module' {
+        Join-path -path $context.BuildRoot -ChildPath $modulePath | should -Exist
+    }
+    
+}
+
+function ThenModuleShouldNotExist
+{
+    $modulePath = Join-path -Path 'Modules' -ChildPath $moduleName
+    it 'should not have the module installed' {
+        Join-path -path $context.BuildRoot -ChildPath $modulePath | should -not -Exist
+    }
+}
+
+function ThenErrorShouldBeThrown
+{
+    param(
+        [string]
+        $errorMessage
+    )
+    it ('should throw an error that matches ''{0}''' -f $errorMessage){
+        $Global:Error | where-object {$_ -match $errorMessage } | Should -not -BeNullOrEmpty
+    }
+}
+Describe 'Get-WhiskeyPowerShellModule.when a valid module is requested without version parameter' {
+    GivenContext
+    GivenModule
+    WhenPowershellModuleIsRan
+    ThenModuleShouldBeInstalled
+}
+
+Describe 'Get-WhiskeyPowerShellModule.when a valid module is requested with version parameter' {
+    GivenContext
+    GivenModule
+    GivenVersion
+    WhenPowershellModuleIsRan
+    ThenModuleShouldBeInstalled
+}
+
+Describe 'Get-WhiskeyPowerShellModule.when an invalid module name is requested' {
+    GivenContext
+    GivenVersion
+    GivenInvalidModule
+    WhenPowershellModuleIsRan
+    ThenErrorShouldBeThrown -errorMessage 'No match was found for the specified search criteria and module name'
+    ThenModuleShouldNotExist
+}
+
+Describe 'Get-WhiskeyPowerShellModule.when called with invalid version' {
+    GivenContext
+    GivenModule
+    GivenInvalidVersion
+    WhenPowershellModuleIsRan
+    ThenErrorShouldBeThrown -errorMessage "Failed to find module Carbon version 0.0.0"
+    ThenModuleShouldNotExist
+}
+
+Describe 'Get-WhiskeyPowerShellModule.when called with missing name' {
+    GivenContext
+    WhenPowershellModuleIsRan
+    ThenErrorShouldBeThrown -errorMessage "Please Add a Name Property for which PowerShell Module you would like to get."
+    ThenModuleShouldNotExist
+}
+
+Describe 'Get-WhiskeyPowerShellModule.when called with clean mode' {
+    GivenContext
+    GivenModule
+    GivenCleanMode
+    WhenPowershellModuleIsRan
+    ThenModuleShouldNotExist
+}
```

Comment:
Add a test that ensures you can download a version with a wildcard:

    Describe 'Get-WhiskeyPowerShellModule.when called with clean mode' {
        GivenContext
        GivenModule 'Carbon'
        GivenVersion "1.*"
        WhenPowershellModuleIsRan
        ThenModuleShouldBeInstalled -AtVersion '1.9.0'
    }

Reply:
Done